test/readrows.ts

```diff
@@ -461,6 +462,45 @@ describe('Bigtable/ReadRows', () => {
     })();
   });
 
+  it.skip('pitfall: should not request full table scan during a retry on a transient error', () => {
+    const TRANSIENT_ERROR_SERVICE: ReadRowsServiceParameters = {
+      chunkSize: CHUNK_SIZE,
+      valueSize: VALUE_SIZE,
+      chunksPerResponse: CHUNKS_PER_RESPONSE,
+      keyFrom: STANDARD_KEY_FROM,
+      keyTo: STANDARD_KEY_TO,
+      deadlineExceededError: true,
+      debugLog,
+    };
+
+    function readRowsWithDeadline() {
+      service.setService({
+        ReadRows: ReadRowsImpl.createService(
+          TRANSIENT_ERROR_SERVICE
+        ) as ServerImplementationInterface,
+      });
+
+      table.createReadStream();
+    }
+    // Spy on the retry mechanism.
+    const retrySpy = sinon.spy(table, 'createReadStream');
```

Comment:
I would replace this with `const requestSpy = sinon.spy(table.bigtable, 'request_')` since this spy is currently useless. We would always expect `createReadStream` to be called exactly once.